bootloader/ubootenv/env.go

```diff
@@ -58,18 +56,39 @@ func writeUint32(u uint32) []byte {
 	return buf.Bytes()
 }
 
-// Create a new empty uboot env file with the given size
-func Create(fname string, size int) (*Env, error) {
+const sizeOfUint32 = 4
+
+func calcHeaderSize(headerFlagByte bool) int {
+	if headerFlagByte {
+		// If uboot uses a header flag byte, header is 4 byte crc + flag byte
+		return sizeOfUint32 + 1
+	}
+	// otherwise, just a 4 byte crc
+	return sizeOfUint32
+}
+
+type CreateOptions struct {
+	HeaderFlagByte bool
+}
+
+// Create a new empty uboot env file with the given size. Passing a nil value
+// for opts is equivalent to passing &CreateOptions{HeaderFlagByte: true}
+func Create(fname string, size int, opts *CreateOptions) (*Env, error) {
 	f, err := os.Create(fname)
 	if err != nil {
 		return nil, err
 	}
 	defer f.Close()
 
+	if opts == nil {
```

Comment:
super-nitpick: I would move this at the start of the function, it's what we do in the rest of the code and this way we never accidently use opts when it might be still nil (we don't right now but we could when we refactor code etc)